Dockerfile

```diff
@@ -1,23 +1,10 @@
-FROM registry.access.redhat.com/ubi9/python-311
-ARG POETRY_VERSION
-RUN pip install --upgrade pip && \
-    pip install poetry==$POETRY_VERSION
+FROM registry.access.redhat.com/ubi9/python-311@sha256:129391e5b291e29a3b24a3fdb281a56d7cb8297bdf8ba92bb5c8581b53084324 AS test
```

Comment:
should we make python 3.12 the default now that the ubi9 image exists or you prefer to leave for a later PR?

Reply:
+1 considering this will only be used for new projects. Existing projects manually migrating to this template can make the decision to upgrade or not

Reply:
Changed in [52abe6d](https://github.com/app-sre/python-project-template/pull/3/commits/52abe6d8eaae85d9e859ac5ebb19e305ef9d9a77)